.github/workflows/build.yml

```
@@ -201,3 +201,65 @@ jobs:
           repo_token: ${{ secrets.GITHUB_TOKEN }}
           tag: ${{ env.ONNXRUNTIME_VERSION }} # ==> github.event.release.tag_name
           file: ${{ env.RELEASE_NAME }}.tgz
+  
+  build-xcframework:
+    needs: build-onnxruntime
+    runs-on: macos-12
+    steps:
+      - name: Generate RELEASE_NAME and ONNXRUNTIME_BASENAME
+        run: |
+          echo "RELEASE_NAME=onnxruntime-xcframework-${{ env.ONNXRUNTIME_VERSION }}" >> $GITHUB_ENV
+          echo "ONNXRUNTIME_BASENAME=libonnxruntime.${{ env.ONNXRUNTIME_VERSION }}.dylib" >> "$GITHUB_ENV"
+

```

おそらく step を書かずに、以下のように job の env で RELEASE_NAME と ONNXRUNTIME_BASENAME を定義できるかな？ と思いました（たぶん動作すると思うのですが、動作する例や情報を見つけられませんでした……）。

```suggestion
    env:
      RELEASE_NAME: onnxruntime-xcframework-${{ env.ONNXRUNTIME_VERSION }}
      ONNXRUNTIME_BASENAME: libonnxruntime.${{ env.ONNXRUNTIME_VERSION }}.dylib
    steps:
```

---

> おそらく step を書かずに、以下のように job の env で RELEASE_NAME と ONNXRUNTIME_BASENAME を定義できるかな？ と思いました（たぶん動作すると思うのですが、動作する例や情報を見つけられませんでした……）。

a1b12e8 で試しましたが、envコンテキスト定義フィールドにおいてenvコンテキストを参照することはできないようでした。

---

> [a1b12e8](https://github.com/VOICEVOX/onnxruntime-builder/commit/a1b12e84d3c6cccbc9a5f5c1a292a70328e44716) で試しましたが、envコンテキスト定義フィールドにおいてenvコンテキストを参照することはできないようでした。

試していただきありがとうございます。参照できなかったとのことで、間違った提案をしてしまいすみませんでした。

（この辺りの仕様を記述したドキュメントを個人的にこれまで見つけられていなかったので、はっきりした知識を持っていませんでした。今回改めて調べたところ、[Context availability](https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability) という項目で、**workflow の特定の場所で使える context の対応表**を見つけたので、メモとして共有しておきます。今回の場合、`jobs.<job_id>.env` で使える context は `github, needs, strategy, matrix, vars, secrets, inputs` とあり、確かに `env` は使えないですね。一方、`jobs.<job_id>.steps.env` 内では `env` context が使えると書いてあり、これと混同していつも混乱してしまっていることが分かりました……。）

---

ENGINEは更にそのenvを他のjobでも流用したいという需要があったのですが、そのときはconfig用のjobを作ったりしました。
https://github.com/VOICEVOX/voicevox_engine/blob/3afd8bf67f767ae0b5c67729bff1e25180675e60/.github/workflows/build.yml#L32
このあたりGithub Actionsはまだ発展途上感ありますね･･･！